torch/package/package_exporter.py

```diff
@@ -202,8 +206,11 @@ def __init__(
         else:  # is a byte buffer
             self.buffer = f
 
-        self.zip_file = torch._C.PyTorchFileWriter(f)
-        self.zip_file.set_min_version(6)
+        self.zip_file = zip_file_reader_type(f)
+        assert isinstance(self.zip_file, TorchScriptPackageZipFileWriter)
+        self.script_module_serializer = torch._C.ScriptModuleSerializer(self.zip_file.zip_file_writer)
+        self.storage_context = self.script_module_serializer.storage_context()
```

Comment:
seems like these torchscript specifics should be moved into the TorchScriptPackageZipFileWriter so it can handle this when torch is present only